utils/debug.c

```diff
@@ -44,24 +56,29 @@ const char *color_enum_or = TERM_COLOR_RESET TERM_COLOR_BOLD "|" TERM_COLOR_RESE
 static const struct color_code {
 	char		code;
 	const char	*color;
+	const char	*html_color;
 } colors[] = {
-	{ COLOR_CODE_RED,	TERM_COLOR_RED },
-	{ COLOR_CODE_GREEN,	TERM_COLOR_GREEN },
-	{ COLOR_CODE_BLUE,	TERM_COLOR_BLUE },
-	{ COLOR_CODE_YELLOW,	TERM_COLOR_YELLOW },
-	{ COLOR_CODE_MAGENTA,	TERM_COLOR_MAGENTA },
-	{ COLOR_CODE_CYAN,	TERM_COLOR_CYAN },
-	{ COLOR_CODE_GRAY,	TERM_COLOR_GRAY },
-	{ COLOR_CODE_BOLD,	TERM_COLOR_BOLD },
+	{ COLOR_CODE_NORMAL,	TERM_COLOR_NORMAL,	HTML_COLOR_NORMAL },
+	{ COLOR_CODE_RESET,	TERM_COLOR_RESET,	HTML_COLOR_RESET },
+	{ COLOR_CODE_RED,	TERM_COLOR_RED,		HTML_COLOR_RED },
+	{ COLOR_CODE_GREEN,	TERM_COLOR_GREEN,	HTML_COLOR_GREEN },
+	{ COLOR_CODE_BLUE,	TERM_COLOR_BLUE,	HTML_COLOR_BLUE },
+	{ COLOR_CODE_YELLOW,	TERM_COLOR_YELLOW,	HTML_COLOR_YELLOW },
+	{ COLOR_CODE_MAGENTA,	TERM_COLOR_MAGENTA,	HTML_COLOR_MAGENTA },
+	{ COLOR_CODE_CYAN,	TERM_COLOR_CYAN,	HTML_COLOR_CYAN },
+	{ COLOR_CODE_GRAY,	TERM_COLOR_GRAY,	HTML_COLOR_GRAY },
+	{ COLOR_CODE_BOLD,	TERM_COLOR_BOLD,	HTML_COLOR_BOLD },
 };
 
 static void color(const char *code, FILE *fp)
 {
 	size_t len = strlen(code);
 
-	if ((fp == logfp && log_color == COLOR_OFF) ||
-	    (fp == outfp && out_color == COLOR_OFF))
-		return;
+	if (format_mode != FORMAT_HTML) {
```

Comment:
Well, I think color and output format is independent.  We might have a html without color..

Reply:
But we can just dump the plain text if someone wants to print without color. So I think this is needed.

Especially, we can't use redirection to save html format into another file without this. This makes output file contain comments in html gray color tag.

Reply:
I think you can set opts->color to yes if format is HTML and user didn't set the color option.

Reply:
Yeah, I can do that. just updated as you mentioned.